### PR TITLE
fix(utils): reject oversized host traces before boolean-trace width hang

### PR DESCRIPTION
**Pull request by ss251**
## Hang

`_reject_boolean_numeric_trace` already rejected hostile nesting deeper than 32, but a wide host list still walked every item (and formatted a unique `name[index]` string) with no value-count bound.

On origin/main `b7bafadd`:

```
compute_running_mean([0.0] * 15_000_000, window_size=2)
```

took **3.442s**. Pointer-repeat list construction is cheap; the cost is the unbounded Python walk before `np.asarray`.

This is occupancy-FREE (`alberta_framework/utils/metrics.py`). It is not leftover INT32/256MiB, not forager/clocks/IA, not JSON nesting, not `jax.tree`, and not a from_config collection rebuild.

## Fix

Cap `_BOOLEAN_TRACE_MAX_NODES = 1_000_000` and reject oversized lists, tuples, object-arrays, index vectors, and metric histories **via `len()` / `.size` before the walk**. Public last-fit remains the 6-step running-mean fixture in `tests/test_continual_metrics.py`. The existing 2000-deep nest still raises `ValueError` for depth, not `RecursionError`.

Oversized pointer-repeat now raises `ValueError` in <0.25s.

## Tests

`tests/test_metrics_boolean_trace_width_hang.py` plus the existing metrics hostile and continual-metrics suites (70 passed).

- AI assistance: yes

- Model(s) used: xai/grok-4.6

- Agent tooling: Grok CLI via manaflow cmux

- Skill path: elizaOS/slopdotcash@6000f7847387c3f046d2c4df071a7c0bb8ddc8d6:skills/contribute-to-asi

- Provenance status: self-reported

AI provider/model: xAI / grok-4.6
Client / agent tooling: Grok CLI via manaflow cmux
Contribution skill revision: elizaOS/slopdotcash@6000f7847387c3f046d2c4df071a7c0bb8ddc8d6:skills/contribute-to-asi
Attribution status: self-reported
— [grok-cli-ss251]

### CODE REVIEW

**Comment by lalalune** [comment]
Superseded by contributor-preserving #2085 after audit. The original production fix is retained there, while the successor replaces the 15M-element/~120 MiB regression allocation and sub-250ms wall-clock assertions with bounded monkeypatched-cap tests that directly cover object/index/nested paths. Closing to prevent duplicate merge; #2085 is the authoritative review lineage.

**Comment by lalalune** [comment]
Superseded by #2085. The successor preserves the contributor implementation, adds the traversal-wide cumulative budget required to close nested/shared amplification, and replaces allocation-heavy timing tests with exact bounded preflight regressions. Independent focused validation: 76 tests, Ruff, and mypy pass.
